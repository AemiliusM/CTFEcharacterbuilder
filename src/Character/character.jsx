export default function Character({ head, middle, bottom }) {
  return (
    <section>
      <article
        aria-label="head"
        className="head"
        style={{
          backgroundImage: `url(./${head}-head.png)`,
          width: 150,
          height: 150,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
        }}
      />
      <article
        aria-label="middle"
        className="middle"
        style={{
          backgroundImage: `url(./${middle}-middle.png)`,
          marginRight: 300,
          width: 200,
          height: 200,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
        }}
      />
      <article
        aria-label="bottom"
        className="bottom"
        style={{
          backgroundImage: `url(./${bottom}-bottom.png)`,
          width: 150,
          height: 150,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
        }}
      />
    </section>
  );
}
