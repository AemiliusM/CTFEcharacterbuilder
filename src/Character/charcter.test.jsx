import { screen, render } from '@testing-library/react';
import Character from './Character.jsx';

it('Should display the Character from initial load', () => {
  const { container } = render(<Character head="baby" middle="strong" bottom="twisted" />);

  const head = screen.getByLabelText(/head/i);
  const torso = screen.getByLabelText(/middle/i);
  const legs = screen.getByLabelText(/bottom/i);

  expect(head).toHaveStyle(`background-image: url(./baby-head.png)`);
  expect(torso).toHaveStyle(`background-image: url(./strong-middle.png)`);
  expect(legs).toHaveStyle(`background-image: url(./twisted-bottom.png)`);
  expect(container).toMatchSnapshot();
});
