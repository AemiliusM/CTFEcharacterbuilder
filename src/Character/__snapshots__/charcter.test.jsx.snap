// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display the Character from initial load 1`] = `
<div>
  <section>
    <article
      aria-label="head"
      class="head"
      style="background-image: url(./baby-head.png); width: 150px; height: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="middle"
      class="middle"
      style="background-image: url(./strong-middle.png); margin-right: 300px; width: 200px; height: 200px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="bottom"
      class="bottom"
      style="background-image: url(./twisted-bottom.png); width: 150px; height: 150px; background-size: cover; background-repeat: no-repeat;"
    />
  </section>
</div>
`;
